worker/backend/spec/spec.go

```diff
@@ -120,6 +129,55 @@ func OciIDMappings(privileged bool, max uint32) []specs.LinuxIDMapping {
 	}
 }
 
+func OciResources(limits garden.Limits) *specs.LinuxResources {
+	var (
+		cpuResources    *specs.LinuxCPU
+		memoryResources *specs.LinuxMemory
+		pidLimit        *specs.LinuxPids
+	)
+	shares := limits.CPU.LimitInShares
+	if limits.CPU.Weight > 0 {
+		shares = limits.CPU.Weight
+	}
+
+	if shares > 0 {
+		cpuResources = &specs.LinuxCPU{
+			Shares: &shares,
+		}
+	}
+
+	memoryLimit := int64(limits.Memory.LimitInBytes)
+	if memoryLimit > 0 {
+		memoryResources = &specs.LinuxMemory{
+			Limit: &memoryLimit,
+			Swap:  &memoryLimit,
+		}
+	}
+
+	maxPids := int64(limits.Pid.Max)
+	if maxPids > 0 {
+		pidLimit = &specs.LinuxPids{
+			Limit: maxPids,
+		}
+	}
+
+	if cpuResources == nil && memoryResources == nil && pidLimit == nil {
+		return nil
+	}
+	return &specs.LinuxResources{
+		CPU:    cpuResources,
+		Memory: memoryResources,
+		Pids:   pidLimit,
+	}
+}
+
+func OciCgroupsPath(basePath, handle string, privileged bool) string {
+	if privileged {
+		return ""
```

Comment:
iiiiiinteeresting

from what I undeerstand, this means that unprivileged containers end up all sharing the "garden tree", while priveleged ones go right into the root of each controller, not sharing what could get enforced for that tree 🤔  

reminds me of what kubernetes does (https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/)

t